### PR TITLE
[zh-cn]: update the translation of Document `open()` method

### PR DESCRIPTION
**Pull request by T34-active**
### Description
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Document/open
### Related issues and pull requests
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/document/open/index.md
* Last commit: https://github.com/mdn/content/commit/41a8b9c9832359d445d136b6d7a8a28737badc6b
https://github.com/mdn/translated-content/pull/20901

### CODE REVIEW

File: files/zh-cn/web/api/document/open/index.md [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+title: Document：open() 方法
 slug: Web/API/Document/open
+l10n:
+  sourceCommit: 41a8b9c9832359d445d136b6d7a8a28737badc6b
 ---
 
 {{APIRef("DOM")}}
 
-**`Document.open()`** 方法打开一个要[写入](/zh-CN/docs/Web/API/Document/write)的文档。
+**`Document.open()`** 方法可打开文档供{{domxref("Document.write", "写入", "", "1")}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/document/open/index.md:10]:
```suggestion
**`Document.open()`** 方法可打开文档以供{{domxref("Document.write", "写入", "", "1")}}。
```

File: files/zh-cn/web/api/document/open/index.md [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
 
 这将会有一些连带的影响。例如：
 
-- 此时已注册到文档、文档中的节点或文档的 window 的所有事件监听器会被清除。
+- 文档内所有节点、文档内或文档窗口上的事件监听器均移除。
```
Comment (yin1999) [inline files/zh-cn/web/api/document/open/index.md:14]: 原翻译更好

File: files/zh-cn/web/api/document/open/index.md [hunk 3]
```diff
@@ -24,60 +26,49 @@ document.open();
...
+## 备注
 
-### 针对 Gecko 的注意事项
+在页面加载后调用 {{domxref("document.write()")}} 时，会自动调用 `document.open()`。
 
-从 Gecko 1.9 开始，这个方法与其他属性一样受到同源策略的控制，若调用会使文档的源产生变化则不可用。
+### 内容安全
 
-从 Gecko 1.9.2 开始，`document.open()` 使用文档的使用的 URI 的[principal](/zh-CN/docs/Security_check_basics)大，而不是从 stack 中取来 principal。因此，你无需再在不可信的文档里调用 {{domxref("document.write()")}} ，包括使用[`wrappedJSObject`](/zh-CN/wrappedJSObject)。关于 principal 的更多信息详见[Security check basics](/zh-CN/Security_check_basics)。
+该方法与其他属性一样受[同源策略](/zh-CN/docs/Web/Security/Same-origin_policy)的限制，如果执行该方法会改变文档的源将不起作用。
```
Comment (yin1999) [inline files/zh-cn/web/api/document/open/index.md:49]: “如果执行该方法会改变文档的源将不起作用”是一个病句

```suggestion
该方法与其他属性一样受[同源策略](/zh-CN/docs/Web/Security/Same-origin_policy)的限制，会改变文档的来源的方法调用不会起作用。
```

File: files/zh-cn/web/api/document/open/index.md [hunk 4]
```diff
@@ -24,60 +26,49 @@ document.open();
...
 
 ## 三个参数的 document.open()
 
-有一个更少人知道且更少被使用的 `document.open()` 的版本，这是{{domxref("Window.open()")}} 的一个别名（前往该页面查看更多）。
+`document.open()` 有一个鲜为人知且很少使用的三参数版本，它是 {{domxref("Window.open()")}} 的别名（详见其页面）。
 
-这种调用，例如在新窗口打开 github.com，把 opener 设为`null`：
+例如，该调用会在新窗口中打开 github.com，并将其打开程序设置为 `null`：
```
Comment (yin1999) [inline files/zh-cn/web/api/document/open/index.md:55]: opener 的翻译来自哪里？

File: files/zh-cn/web/api/document/open/index.md [hunk 5]
```diff
@@ -24,60 +26,49 @@ document.open();
...
-document.open(type, replace)
+```js
+document.open(type, replace);
 ```
 
-`type`指定了所需写入的数据的 MIME 类型，`replace`（如有设置，值为一个字符串“replace”）指定了新文档的历史写入会代替现有的例如写入。
+`type` 指定要写入的数据的 MIME 类型（例如 `text/html`），如果设置了 replace（即 `"replace"` 字符串），新文档历史记录会替换当前写入文档的记录。
 
-这种形式现在已经弃用；它不会抛出错误，但会直接调用`document.open()`（相当于无参数形式的调用）。这种历史写入替换行为现在一定会发生。
+这种形式现已过时；它不会抛出错误，而是直接转发到 `document.open()`（即相当于没有参数运行它）。历史替换行为现在总是会发生。
```
Comment (yin1999) [inline files/zh-cn/web/api/document/open/index.md:71]: 原翻译不是更好吗

```suggestion
这种形式现已过时；它不会抛出错误，而是直接转发到 `document.open()`（相当于无参数形式的调用）。历史替换行为现在总是会发生。
```

File: files/zh-cn/web/api/document/open/index.md (hunk 4 above)
Comment (T34-active) [reply files/zh-cn/web/api/document/open/index.md:55]: https://github.com/mdn/content/blob/main/files/en-us/web/api/document/open/index.md?plain=1#L62

File: files/zh-cn/web/api/document/open/index.md (hunk 2 above)
Comment (T34-active) [inline files/zh-cn/web/api/document/open/index.md:14]:
```suggestion
- 此时已注册到文档、文档中的节点或文档的 window 的所有事件监听器会被清除。
```

File: files/zh-cn/web/api/document/open/index.md (hunk 4 above)
Comment (yin1999) [reply files/zh-cn/web/api/document/open/index.md:55]: 我说的是，为啥把 opener 翻译为“打开程序”

Comment (T34-active) [reply files/zh-cn/web/api/document/open/index.md:55]: 看错了，这里不能翻译的

File: files/zh-cn/web/api/document/open/index.md [hunk 6]
```diff
@@ -24,60 +26,49 @@ document.open();
...
-从 Gecko 1.9.2 开始，`document.open()` 使用文档的使用的 URI 的[principal](/zh-CN/docs/Security_check_basics)大，而不是从 stack 中取来 principal。因此，你无需再在不可信的文档里调用 {{domxref("document.write()")}} ，包括使用[`wrappedJSObject`](/zh-CN/wrappedJSObject)。关于 principal 的更多信息详见[Security check basics](/zh-CN/Security_check_basics)。
+该方法与其他属性一样受[同源策略](/zh-CN/docs/Web/Security/Same-origin_policy)的限制，会改变文档的来源的方法调用不会起作用。
 
 ## 三个参数的 document.open()
 
-有一个更少人知道且更少被使用的 `document.open()` 的版本，这是{{domxref("Window.open()")}} 的一个别名（前往该页面查看更多）。
+`document.open()` 有一个鲜为人知且很少使用的三参数版本，它是 {{domxref("Window.open()")}} 的别名（详见其页面）。
 
-这种调用，例如在新窗口打开 github.com，把 opener 设为`null`：
+例如，该调用会在新窗口中打开 github.com，并将其打开程序设置为 `null`：
```
Comment (T34-active) [inline files/zh-cn/web/api/document/open/index.md:55]:
```suggestion
例如，该调用会在新窗口中打开 github.com，并将其 opener 设置为 `null`：
```

**Review by yin1999 — APPROVED** [review]

**Pull request merged**